UnitTests/DictionariesTest.cs

```diff
@@ -37,5 +38,74 @@ public void AppendNull()
             Assert.Equal(src["one"], "1");
             Assert.Equal(src["two"], "2");
         }
+
+        [Fact]
+        public void MergeHeaders_CalledWithOnlyNulls_ReturnsEmptyDictionary()
+        {
+            //given
+
+
+            //when
+            var ret = Dictionaries.MergeHeaders(null, null);
+
+            //then 
+            Assert.NotNull(ret);
+            Assert.Empty(ret);
+        }
+
+        [Fact]
+        public void MergeHeaders_MergeNullDictionary_ReturnsCopyOfOriginal()
+        {
+            //given
+            var src = new Dictionary<string, object> { { "one", "1" }, { "two", "2" } };
+
+            //when
+            var ret = Dictionaries.MergeHeaders(src, null);
+
+            //then
+            Assert.NotSame(ret, src);
+            Assert.Equal(2, ret.Count);
+            Assert.Equal("1", ret["one"]);
+            Assert.Equal("2", ret["two"]);
+        }
+
+        [Fact]
+        public void MergeHeaders_MergeUniqueKeys_ReturnsExpectedunion()
+        {
+            //given
+            var dict1 = new Dictionary<string, object> { { "one", "1" }, { "two", "2" } };
+            var dict2 = new Dictionary<string, object> { { "three", "3" }, { "four", "4" } };
+            var dict3 = new Dictionary<string, object> { { "five", 5 }, { "six", 6 } };
+
+            //when
+            var ret = Dictionaries.MergeHeaders(dict1, dict2, dict3);
+
+            //then
+            Assert.Equal(ret.Count, 6);
+            Assert.Equal("1", ret["one"]);
+            Assert.Equal("2", ret["two"]);
+            Assert.Equal("3", ret["three"]);
+            Assert.Equal("4", ret["four"]);
+            Assert.Equal(5, ret["five"]);
+            Assert.Equal(6, ret["six"]);
+        }
+
+
+        [Fact]
+        public void MergeHeaders_MergeNonUniqueKeys_ThrowsArgumentException()
+        {
+            //given
+            var dict1 = new Dictionary<string, object> { { "one", "1" }, { "two", "2" } };
+            var dict2 = new Dictionary<string, object> { { "three", "3" }, { "four", "4" } };
+            var dict3 = new Dictionary<string, object> { { "five", 5 }, { "one", 6 } };
+
+            //when
+            var exception = Record.Exception(() => Dictionaries.MergeHeaders(dict1, dict2, dict3));
```

Comment:
Have question here, it seems `.MergeHeaders()` will throw exception if duplicate keys found?  

It seems to be different from current behavior with `enc`, `alg` headers we discussed before. It current duplicates are simply ignored without failing.

Reply:
Hi - yes you are correct the behaviour has changed to be a throw rather than a silent ignore.
Personally I preferred the throw rather than user not realising their request was ignored.
However if you want the previous behaviour maintained I could re-work it to have the previous behaviour? Should I make it do a silent ignore?

Reply:
Well, I'd may be prefer to throw it too, but on the other side there are existing users where given change can unexpectedly break applications. So if you don't mind to update it, would appreciate if we can keep compatible behavior.

Reply:
Hi, sorry even though I said I'd change the behaviour I timed-out getting a backward compatible behaviour back in, within time was allowed.
It's more complicated with 3 (or 4) different sources of headers now.
(Note for example Golang errors on dups... https://github.com/latchset/jwcrypto/blob/master/jwcrypto/jwe.py)

Do you think the change in behaviour is a deal-breaker for the PR?

Reply:
No, i don't think it is show stopper. Let me think for a while and i get back to you (tomorrowish)

Reply:
Ok, probably let's push it with new semantics, can you please add notes to `breaking changes` section here: https://github.com/dvsekhvalnov/jose-jwt#important-upgrade-notes ?

Reply:
Sure - I have put in a note there.

Reply:
Looks good to me. Thanks.